pybamm/models/submodels/particle_cracking/crack_propagation.py

```diff
@@ -0,0 +1,80 @@
+import pybamm
+from .base_cracking import BaseCracking
+import numpy as np
+
+
+class CrackPropagation(BaseCracking):
+    """cracking behaviour in electrode particles.
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+    domain : str
+        The domain of the model either 'Negative' or 'Positive'
+    requiring the radius, average concantration, surface concantration
+
+    Ref for the crack model: Deshpande, R., Verbrugge, M., Cheng, Y. T.,
+    Wang, J., & Liu, P. (2012). Battery cycle life prediction with coupled
+    chemical degradation and fatigue mechanics. Journal of the Electrochemical
+    Society, 159(10), A1730.
+    """
+
+    def __init__(self, param, domain):
+        super().__init__(param, domain)
+
+    def get_fundamental_variables(self):
+        l_cr = pybamm.Variable(
+            self.domain + " particle crack length",
+            domain=self.domain.lower() + " electrode",
+        )
+        return self._get_standard_variables(l_cr)
+
+    def get_coupled_variables(self, variables):
+        variables.update(self._get_mechanical_results(variables))
+        if self.domain == "Negative":
+            l_cr_scale = self.param.l_cr_n_0
+            Eac_cr = self.param.Eac_cr_n
+        else:
+            l_cr_scale = self.param.l_cr_p_0
+            Eac_cr = self.param.Eac_cr_p
+        stress_t_surf = variables[
+            self.domain + " particle surface tangential stress [Pa]"
+        ]
+        l_cr = variables[self.domain + " particle crack length"]
+        R_const = self.param.R_const
+        Delta_T = self.param.Delta_T
+        T_dim = variables[self.domain + " electrode temperature [K]"]
+        k_cr = self.param.k_cr * pybamm.exp(
+            Eac_cr / R_const * (1 / T_dim - 1 / self.param.T_ref)
+        )  # cracking rate with temperature dependence
+        # # compressive stress will not lead to crack propagation
+        dK_SIF = (
+            stress_t_surf
+            * self.param.b_cr
+            * pybamm.Sqrt(np.pi * l_cr * l_cr_scale)
+            * (stress_t_surf >= 0)
+        )
+        dl_cr = (
+            k_cr * pybamm.Power(dK_SIF, self.param.m_cr) / self.param.t0_cr / l_cr_scale
+        )
+        variables.update(
+            {
+                self.domain + " particle cracking rate": dl_cr,
```

Comment:
I'm a bit confused by what is dimensional and what isn't in these models. I think `dl_cr` should be dimensionless (since the model is in dimensionless form), but the definition depends of `stress_t_surf` which has units [Pa]

Reply:
I would have expected that the models could all be written in terms of the dimensionless parameters you defined, and the dimensional scales only turn up in `base_cracking` to turn the non-dim variables into their dimensional version

Reply:
so I guess in dimensional terms `dl_cr_dim` should have units [m.s-1] so the that equation is `d_l_cr_dim/d_t_dim = dl_cr_dim` I'm not sure what units the individual things in dl_cr end up having, but when non-dimensional i'd expect a factor l_cr_ref/tau on the left. I think this comes back to me not understanding why `self.t0_cr = 3600 / self.timescale`. I don't get where the 3600 comes from? there should be some timescale inherent in the cracking equation, related to the crack rate?

Reply:
Agree with you that it is better to have dimensionless equation in pybamm, but the dimension less version for this equation is not simple. I need to think about what are the right dimentionless expressions. 

It defines a time constant `self.t0_cr` for the time per cycle, transfering the cracking rate from per cycle to per time. I though the time in pybamm is in second by default, so I put the number `3600` for 1 hour, which depends on how to set the time unit.    `

Reply:
the time in pybamm is scaled with `self.timescale` which for li-ion (and lead-acid) is the discharge timescale `tau_discharge`. I would expect you will get a ratio of discharge timescale to cracking timescale in the dimensionless equation. could you point me to the reference for the cracking model please? maybe we can take a look together to make sure the way it is scaled is consistent with the other models in pybamm

Reply:
Have a look at the Equation 9 in the reference below. Here I made dl/dN to dl/dt*dt/dN, so there is timescale involved. I would use the same normalisation for time, otherwise it will be very confusing.

  [Deshpande, R., Verbrugge, M., Cheng, Y. T., Wang, J., & Liu, P. (2012). Battery cycle life prediction with coupled chemical degradation and fatigue mechanics. Journal of the Electrochemical Society, 159(10), A1730.](https://iopscience.iop.org/article/10.1149/2.049210jes/meta)

Reply:
Thanks, I'll take a proper look later. I didn't realise that the equation was defined per cycle - not clear how to define the time for a single cycle presumably cycle time changes as the cracks grow? and what happens when you cycle at different rates? or if you want to model cracking but you aren't doing full cycles? (sorry for all the questions)

Reply:
No worries, those are good questions. The equation was defined per cycle for a fatigue crack model, because it needs a number of cycles for the damage to accumulate. The cracking rate is so slow that the cracking rate is almost not changed within one cycle. 

For your questions, 1) the constants are obtained from experiment, so it depends on how experiment degradation data were processed. It can make `t0_cr` rate dependent, which I can modify, but this introduces an extra variable in the equation. 2) The magnitude of stress will change for different rates, which affect the cracking rate. 3) If not cycling in full, the time integration of dl/dt will account for this, which is why dl/dt is used instead of dl/dN.